deployer/src/handlers/mod.rs

```diff
@@ -719,9 +730,22 @@ async fn logs_websocket_handler(
                 "failed to get backlog of logs"
             );
 
-            let _ = s
-                .send(ws::Message::Text("failed to get logs".to_string()))
-                .await;
+            if error.code() == tonic::Code::Unavailable
+                && error.metadata().get("x-ratelimit-limit").is_some()
+            {
+                let _ = s
+                    .send(ws::Message::Text(
+                        Error::RateLimited(
+                            "your application is producing too many logs, any interaction with the shuttle logger service will be rate limited"
```

Comment:
We should improve this wording in the same way we decide to improve the earlier wording